core/crypto/src/key_conversion.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{signature, vrf};
+use curve25519_dalek::edwards::{CompressedEdwardsY, EdwardsPoint};
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use digest::Digest;
+use sha2::Sha512;
+use std::mem::transmute;
+
+pub fn convert_public_key(key: &signature::ED25519PublicKey) -> Option<vrf::PublicKey> {
+    let ep: EdwardsPoint = CompressedEdwardsY::from_slice(&key.0).decompress()?;
+    // All properly generated public keys are torsion-free. RistrettoPoint type can handle some values that are not torsion-free, but not all.
+    if !ep.is_torsion_free() {
+        return None;
+    }
+    // Unfortunately, dalek library doesn't provide a better way to do this.
+    let rp: RistrettoPoint = unsafe { transmute(ep) };
+    Some(vrf::PublicKey(rp.compress().to_bytes(), rp))
+}
+
+pub fn convert_secret_key(key: &signature::ED25519SecretKey) -> vrf::SecretKey {
+    let h = Sha512::default().chain(&key.0[..32]).result();
+    let mut s = *array_ref!(h.as_slice(), 0, 32);
+    s[0] &= 248;
+    s[31] &= 63;
+    s[31] |= 64;
+    vrf::SecretKey::from_scalar(Scalar::from_bytes_mod_order(s))
+}
```

Comment:
seems like we can keep the magic inside `ed25519_dalek`:
```
pub fn convert_secret_key(key: &signature::ED25519SecretKey) -> vrf::SecretKey {
    let expanded = ed25519_dalek::ExpandedSecretKey::from(
        &ed25519_dalek::SecretKey::from_bytes(&key.0[..32]).unwrap(),
    );
    let bytes: [u8; 32] = expanded.to_bytes()[..32].try_into().unwrap();
    vrf::SecretKey::from_scalar(Scalar::from_bytes_mod_order(bytes))
}
```

Reply:
Nice idea! I've updated my code to do that.